### PR TITLE
LB-786: Upgrade SocketIO client and server libraries

### PR DESCRIPTION
**Pull request by amCap1712**
Upgrade Flask-SocketIO to 5.x, it requires the frontend socket-io.client library to be also upgraded to 3.x at least. Things seemed to work, hence socket-io.client has been upgraded to 4.x. Types are also provided by the socket-io.client, hence remove the deprecated package @types/docket-io.client package.

### CODE REVIEW

**Review by alastair — COMMENTED** [review]
looks like there's still a failing frontend test, but otherwise if you tested the backend implementation and it works then that's fine by me!

**Comment by amCap1712** [comment]
Right, I tested on test.lb and it worked. I was unable to fix the test though, I hope @MonkeyDo can help :)

**Review by MonkeyDo — APPROVED** [review]
Tests fixed !

File: package.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     "react-select": "^3.1.0",
     "react-sortablejs": "^6.0.0",
     "react-youtube": "^7.12.0",
-    "socket.io-client": "^2.2.0",
+    "socket.io-client": "^4.1.2",
```
Comment (MonkeyDo) [inline package.json:66]: This introduced a regression for older browsers —such as CatQuest's FF44 — either the newer version of this library or one of its dependencies isn't compiled to ES5.
I'll open a PR to fix it.

Comment (amCap1712) [reply package.json:66]: Oh! Thanks!

Comment (MonkeyDo) [reply package.json:66]: For posterity -> #1524

**Pull request merged**